GVRf/Framework/src/org/gearvrf/GVRSceneObject.java

```diff
@@ -239,15 +239,18 @@ public String getName() {
     }
 
     /**
-     * Set the (optional) name of the object.
+     * Set the (optional) name of the object. Scene object names are not needed:
+     * they are only for the application's convenience.
      * 
-     * Scene object names are not needed: they are only for the application's
-     * convenience.
-     * 
-     * @param name
-     *            Name of the object.
+     * @param name Name of the object.
+     * @throws IllegalArgumentException if input parameter is null or empty.
      */
     public void setName(String name) {
+        if (name == null || name.trim().isEmpty()) {
```

Comment:
I agree that all-whitespace names would be easy to confuse and/or hard to spot in a log; I don’t see how it is GVRF’s responsibility to guard against that. The point of these parameter checks is to avoid native crashes, not to keep app developers from doing something that seems foolish to us but may actually have some sound reason in the context of the app.

From: jason2kim [mailto:notifications@github.com]
Sent: Wednesday, May 06, 2015 12:13 PM
To: Samsung/GearVRf
Cc: Jon Shemitz
Subject: Re: [GearVRf] fix for some similar issues of, (#31)


In GVRf/Framework/src/org/gearvrf/GVRSceneObject.java<https://github.com/Samsung/GearVRf/pull/31#discussion_r29791601>:

>       */

>      public void setName(String name) {

> +        if (name == null || name.trim().isEmpty()) {

No, with map, I didn’t mean anything related to *. I was saying possible uses of maps in the applications themselves, where they will use the name as a key with logical names.

—
Reply to this email directly or view it on GitHub<https://github.com/Samsung/GearVRf/pull/31/files#r29791601>.